lib/addons/addon-schema.js

```diff
@@ -0,0 +1,34 @@
+const joi = require('joi');
+const { nameType } = require('../routes/admin-api/util');
+const { tagTypeSchema } = require('../services/tag-type-schema');
+
+const addonDefinitionSchema = joi.object().keys({
+    name: nameType,
+    displayName: joi.string(),
+    documentationUrl: joi.string().uri({ scheme: [/https?/] }),
+    description: joi.string().allow(''),
+    parameters: joi
+        .array()
+        .optional()
+        .items(
+            joi.object().keys({
+                name: joi.string().required(),
+                displayName: joi.string().required(),
+                type: joi.string().required(),
+                description: joi.string(),
+                required: joi.boolean().default(false),
+            }),
+        ),
+    events: joi
+        .array()
+        .optional()
+        .items(joi.string()),
+    tags: joi
```

Comment:
```suggestion
    tagTypes: joi
```
As was already discussed earlier in the PR